node/directives/checkUserAccess.ts

```diff
@@ -3,6 +3,63 @@ import type { GraphQLField } from 'graphql'
 import { defaultFieldResolver } from 'graphql'
 import { SchemaDirectiveVisitor } from 'graphql-tools'
 
+export async function checkUserOrAdminTokenAccess(
+  ctx: Context,
+  operation?: string
+) {
+  const {
+    vtex: { adminUserAuthToken, storeUserAuthToken, logger },
+    clients: { identity, vtexId },
+  } = ctx
+
+  if (!adminUserAuthToken && !storeUserAuthToken) {
+    logger.warn({
+      message: `CheckUserAccess: No admin or store token was provided for ${operation}`,
+      operation,
+    })
+    throw new AuthenticationError('No admin or store token was provided')
+  }
+
+  if (adminUserAuthToken) {
+    try {
+      await identity.validateToken({ token: adminUserAuthToken })
+    } catch (err) {
+      logger.warn({
+        error: err,
+        message: `CheckUserAccess: Invalid admin token for ${operation}`,
```

Comment:
My only remaining suggestion here is to remove this `${operation}` text from the log message. You already have the operation as one of the fields being logged here. It's a good practice to keep log messages static, without variables within them, using just fields instead. That helps analyze and aggregate them all, and you can use the field when you want to focus on the the differences between different operations. You'll also won't lose historic log data for these log messages, you'll just be adding a field to it.

Reply:
Same for the other log messages in this file.

Reply:
Yes I know, but just reminder that is IO app, so it isn't easy to put a new field to index. If everybody wants to put a new field there is a similar scenario like [that](https://vtex.slack.com/archives/C02MGEZ392R/p1700681542688149?thread_ts=1700677521.753579&cid=C02MGEZ392R).